### DISCUSSION TITLE
隐藏分享里的头像栏或via memos

### DISCUSSION THREAD

**Discussion by Coisini-H** [usememos/memos#2962] [Show and tell]
一开始我只是打算修改via memos文字，想改成自己的网站名字，结果我改成功，就出了一个隐藏的想法

原图：
![QQ截图20240216001408](https://github.com/usememos/memos/assets/52484657/96c8075e-360b-4bfa-82b4-a1fe33492241)

隐藏via memos后效果：
![%7B810A9830-ACAC-41e1-989B-A7BC4D6A744D%7D](https://github.com/usememos/memos/assets/52484657/28644b58-085f-4b00-8560-94f562f0fe1e)

隐藏头像栏效果：
![QQ截图20240216001450](https://github.com/usememos/memos/assets/52484657/b8303093-fff4-4017-99d5-0676d4309059)

隐藏via memos css代码：
```
body .flex.flex-row.justify-between.items-center > .text-gray-500.dark\:text-gray-400 {
  display: none;
}
```

隐藏头像栏css代码
```
body .flex.flex-row.justify-between.items-center.w-full.bg-gray-100.dark\:bg-zinc-900.py-4.px-6 {
  display: none;
}
```

如果有大佬会修改via memos替换成网站的名字，可以评论告诉我，虽然是一个小改动。

**Comment by forrany** [comment]
把以下带面中的webName改成你想要的内容就好

因为memos没有给出弹窗或者click的回调函数，所以只能使用observer去监测dom的变化（因为分享的弹窗是append在body上的，这个可以在[源码](https://github1s.com/usememos/memos/blob/main/web/src/components/Dialog/BaseDialog.tsx#L73)中看到）, 获取了targetNode后，直接在dom上修改内容。

最新版测试没问题，可以试一下

```js
const webName = 'xx'; // 要修改的网站名称

// 定义需要观察DOM变化的目标节点
const targetNode = document.querySelector('body');

// 观察者配置选项
const observerOptions = {
  childList: true,
  subtree: true // 需要观察子节点变化，包括后代所有节点的变化
};

// 初始化一个MutationObserver
const observer = new MutationObserver((mutationList, observer) => {
  // 使用Find来检查是否存在需要更改名称的元素，如果存在则更改
  const checkAndModifyName = () => {
    const element = document.querySelector('span.text-gray-500.dark\\:text-gray-400');
    if (element) {
      element.textContent = webName; // 使用 textContent 而非 innerHTML 更安全、更快
    }
  };

  // 当观察到子元素列表发生变化时
  for (const mutation of mutationList) {
    if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
      checkAndModifyName();
      break; // 如果已找到并修改，则不需要检查其他变化
    }
  }
});

// 启动观察者来观察目标节点
observer.observe(targetNode, observerOptions);
```

**Reply by Coisini-H** [reply]
谢谢大佬，太感谢了！！
